lib/src/cupertino/cupertino_popover_menu.dart

```diff
@@ -235,21 +268,42 @@ class RenderPopover extends RenderShiftedBox {
     markNeedsPaint();
   }
 
+  /// Returns the size which must be reserved to display the arrow.
+  ///
+  /// When [extendAndClipContentOverArrow] is `true`, the [child] can take the space that is reserved for the arrow.
+  /// During paint, the [child] is clipped using the background shape.
+  ///
+  /// When [allowHorizontalArrow] is `false`, the reserved width is always zero.
+  Size get _reservedSizeForArrow => Size(
+        (allowHorizontalArrow ? arrowLength * 2 : 0),
+        (extendAndClipContentOverArrow ? 0.0 : arrowLength * 2),
+      );
+
+  /// Returns the background shape offset from the top left corner.
+  Offset get _shapeOffset => Offset(
+        (allowHorizontalArrow ? arrowLength : 0),
+        arrowLength,
+      );
+
   Offset _contentOffset = Offset.zero;
 
   bool _showDebugPaint = false;
 
   late Paint _backgroundPaint;
 
+  final LayerHandle<ClipPathLayer> _clipPathLayer = LayerHandle<ClipPathLayer>();
```

Comment:
Should we be instantiating this layer here? Should this layer be nullable, instead?

Also, please comment what this is.

Reply:
The `RenderObject`'s layer is also instantiated this way.